client/src/partials/cash/modals/receipt.modal.js

```diff
@@ -0,0 +1,118 @@
+angular.module('bhima.controllers')
+.controller('CashReceiptModalController', CashReceiptModalController);
+
+CashReceiptModalController.$inject = [
+  'uuid', 'patientUuid', '$uibModalInstance', 'CashService', 'ProjectService',
+  'EnterpriseService', '$q', 'CashboxService', 'UserService', 'ExchangeRateService',
+  'Patients', 'CurrencyService'
+];
+
+/**
+ * @module cash/modals/CashReceiptModalController
+ *
+ * @description This controller is responsible for displaying a receipt for a
+ * cash payment made from the auxillary cash box.
+ */
+function CashReceiptModalController(uuid, patientUuid, ModalInstance, Cash, Projects, Enterprises, $q, Cashboxes, Users, Exchange, Patients, Currencies) {
+  var vm = this;
+
+  // bind close modal method
+  vm.cancel = ModalInstance.dismiss;
+  vm.isEnterpriseCurrency = isEnterpriseCurrency;
+  vm.fmtCurrency = fmtCurrency;
+
+  // bind data
+  vm.loading = false;
+
+  /** generic error handler temporarily */
+  function handler(error) {
+    vm.error = error;
+  }
+
+  /** fired to start up the receipt module */
+  function startup() {
+
+    // turn loading indicator on
+    toggleLoadingIndicator();
+
+    Cash.read(uuid).then(function (receipt) {
+
+      // bind the receipt
+      vm.receipt = receipt;
+
+      return $q.all([
```

Comment:
It has been discussed that this many HTTP requests could be well grouped on the server and then sent to the service in one request. This works for now however just a comment to remind about the API design.